### PR TITLE
Remove malformed empty page files on access

### PR DESCRIPTION
**Pull request by dbw9580**
### What changes are proposed in this pull request?

Remove empty pages when they are accessed by the get page methods. Empty pages except for the vert first page of a file are most likely malformed and are a result of a failed put attempt.

### Why are the changes needed?

Reading empty pages causes 0 bytes to be read, and in some cases causing an infinite loop in the caller. When used with `getDataFileChannel`, it results in the channel to be created multiple times for a given empty page, because the channels are always 0 sized.

### Does this PR introduce any user facing changes?

No.

### CODE REVIEW

**Review by JiamingMai — APPROVED** [review]
LGTM

**Review by beinan — APPROVED** [review]
I'm good for the fix, but I think we still need to figure out the root cause

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java [hunk 1]
```diff
@@ -116,6 +120,11 @@ public int get(PageId pageId, int pageOffset, int bytesToRead, ReadTargetBuffer
         bytesRead += bytes;
         bytesLeft -= bytes;
       }
+      if (bytesRead == 0) {
+        // no bytes have been read at all, but the requested length > 0
+        // this means the file is empty
+        return -1;
```
Comment (dbw9580) [inline dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java:126]: log a warning when this happens

File: dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java [hunk 2]
```diff
@@ -205,13 +214,29 @@ public DataFileChannel getDataFileChannel(
...
     if (!pageFile.exists()) {
       throw new PageNotFoundException(pagePath.toString());
     }
 
     long fileLength = pageFile.length();
+    if (fileLength == 0 && pageId.getPageIndex() > 0) {
+      // pages other than the first page should always be non-empty
+      // remove this malformed page
+      try {
+        Files.deleteIfExists(pagePath);
```
Comment (dbw9580) [inline dora/core/client/fs/src/main/java/alluxio/client/file/cache/store/LocalPageStore.java:230]: log a warning

**Comment by dbw9580** [comment]
alluxio-bot, merge this please

**Pull request merged**